rom/dev/src/flow/cold_reset/fw_processor/mod.rs

```diff
@@ -268,7 +270,7 @@ impl FirmwareProcessor {
         dma: &mut Dma,
         env: &mut KatsEnv,
         persistent_data: &mut PersistentData,
-    ) -> CaliptraResult<(ManuallyDrop<MailboxRecvTxn<'a>>, u32)> {
+    ) -> CaliptraResult<(Option<ManuallyDrop<MailboxRecvTxn<'a>>>, u32)> {
```

Comment:
We should update the documentation for this function to indicate that the MailboxRecvTxn is returned only in passive mode.

And similarly we should update the other functions to indicate that the txn should be None if in subsystem mode, since we are removing the explicit check for subsystem mode now.